src/Spice86.Core/Emulator/Devices/Timer/Pit8254Counter.cs

```diff
@@ -187,17 +247,102 @@ public void WriteReloadValueByte(byte value) {
+                case PitMode.SquareWave:
+                case PitMode.SquareWaveAlias:
+                case PitMode.SoftwareStrobe:
+                    // These modes don't count when gate is disabled
+                    shouldCount = IsGateEnabled;
+                    break;
+            }
+            
+            if (!shouldCount) {
+                return true; // Activation processed but counting inhibited
+            }
+            
             CurrentCount--;
+            
+            // Handle counter reaching zero based on the mode
+            if (CurrentCount == 0) {
+                switch (CurrentPitMode) {
+                    case PitMode.InterruptOnTerminalCount:
+                        // Mode 0: Output goes high when counter reaches terminal count
+                        OutputState = OutputStatus.High;
+                        // Counter stops at zero
+                        break;
+                        
+                    case PitMode.OneShot:
+                        // Mode 1: One-shot - output goes low until counter reaches zero, then high
+                        OutputState = OutputStatus.High;
+                        // Counter stops at zero
+                        break;
+                        
+                    case PitMode.RateGenerator:
+                    case PitMode.RateGeneratorAlias:
+                        // Mode 2: Rate generator - output goes low for one clock cycle when counter reaches zero
+                        OutputState = OutputStatus.Low;
+                        // Reload the counter
+                        CurrentCount = ReloadValue;
+                        // Output should quickly go back high
+                        OutputState = OutputStatus.High;
+                        break;
+                        
+                    case PitMode.SquareWave:
+                    case PitMode.SquareWaveAlias:
+                        // Mode 3: Square wave - output toggles between high and low
+                        OutputState = OutputState == OutputStatus.High ? OutputStatus.Low : OutputStatus.High;
+                        // Reload the counter
+                        CurrentCount = ReloadValue;
+                        break;
+                        
+                    case PitMode.SoftwareStrobe:
+                        // Mode 4: Software triggered strobe - output goes low when counter reaches zero
+                        OutputState = OutputStatus.Low;
+                        // Reload the counter
+                        CurrentCount = ReloadValue;
+                        // Output quickly goes back high
+                        OutputState = OutputStatus.High;
+                        break;
+                        
+                    case PitMode.HardwareStrobe:
+                        // Mode 5: Hardware triggered strobe - similar to mode 4 but hardware triggered
+                        OutputState = OutputStatus.Low;
```

Comment:
Is it relevant to set it low then high in our case? Is there anything watching the change?

Reply:
Implemented with delays now.